packages/@lwc/integration-karma/test/accessibility/non-standard-aria-props/index.spec.js

```diff
@@ -42,7 +42,9 @@ if (process.env.ENABLE_ARIA_REFLECTION_GLOBAL_POLYFILL) {
                                 elm.getProp(prop);
                             }).not.toLogWarningDev();
 
-                            expect(dispatcher).not.toHaveBeenCalled();
+                            expect(dispatcher).not.toHaveBeenCalledWith(
+                                'NonStandardAriaReflection'
+                            );
```

Comment:
It looks like it's going to be painful to update our tests to account for every possible type of log. This could also get hairier in the future as we add more log/report types.

What do you think about adding an optional options argument to `attachDispatcher` that allows for filtering on only the logs we're interested in? That way, each test can just focus on only the logs it cares about.

https://github.com/salesforce/lwc/blob/6e75634e16a21c500d5e3f45400d1db3da3f5346/packages/%40lwc/engine-core/src/framework/reporting.ts#L82

If omitted, the property bag would results in nothing being filtered.

Reply:
I was originally thinking to just update the tests to be more explicit and narrower in scope about what they check. If we narrow their scope - `no logs about NonStandardAriaReflection` - rather than just a general `.not.toHaveBeenCalled()`, we should avoid the need to have to account for every possible type of log in the future and avoid the extra hair.

That said, it might be nice as a feature to let the dispatcher filter on only the log types we're interested in. Given that this is meant to be used at the app level though, it feels weird to have this functionality without the ability to attach different dispatchers for different log types (unless this is mainly a test-only API?). Would we add that ability too, or leave it as 1 dispatcher per app?

Reply:
> I was originally thinking to just update the tests to be more explicit and narrower in scope

Yep, this is exactly the issue I'm looking to solve!

> it feels weird to have this functionality without the ability to attach different dispatchers for different log types (unless this is mainly a test-only API?). Would we add that ability too, or leave it as 1 dispatcher per app?

I would lean towards YAGNI – this is basically a test-only API, so having one dispatcher at a time is fine.

Alternatively, we could go even leaner and just update the tests to do something like:

```js
dispatcher = jasmine.createSpy();
reportingControl.attachDispatcher((id, payload) => {
  if (id === ReportingEventId.NonStandardAriaReflection) {
    dispatcher(id, payload)
  }
});
```

Maybe since this is test-only, we should go with this approach instead? If it's too much boilerplate for each test, we can shove it into `test-utils`.

Reply:
:+1: Yeah, assigning a custom dispatcher in the test seems like the most straightforward way to make the change for the entire test. If we do it in `test-utils`, that would effectively be implementing the ability to filter on only the logs we're interested in, as a test-only API.